atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/lock/IsolatedProcessCorrectnessCheckerTest.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.jepsen.lock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.jepsen.CheckerResult;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.utils.TestEventUtils;
+
+public class IsolatedProcessCorrectnessCheckerTest {
+    private static final int process1 = 1;
+
+    @Test
+    public void shouldSucceedOnNoEvents() {
+        CheckerResult result = runIsolatedProcessRefreshSuccessChecker(ImmutableList.<Event>of());
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
+    }
+
+    @Test
+    public void onlyLocksSucceeds() {
+        ImmutableList<Event> eventList = ImmutableList.<Event>builder()
+                .add(TestEventUtils.invokeLock(0, process1))
+                .add(TestEventUtils.lockSuccess(1, process1))
+                .add(TestEventUtils.invokeLock(2, process1, "alternate_lock"))
+                .add(TestEventUtils.lockFailure(3, process1))
+                .build();
+        CheckerResult result = runIsolatedProcessRefreshSuccessChecker(eventList);
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
+    }
+
+    @Test
+    public void cannotRefreshWithoutLock() {
+        ImmutableList<Event> eventList = ImmutableList.<Event>builder()
+                .add(TestEventUtils.invokeRefresh(0, process1))
+                .add(TestEventUtils.refreshSuccess(1, process1))
+                .add(TestEventUtils.invokeRefresh(2, process1, "alternate_lock"))
+                .add(TestEventUtils.refreshFailure(3, process1))
+                .build();
+        CheckerResult result = runIsolatedProcessRefreshSuccessChecker(eventList);
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(eventList.get(1), eventList.get(1));
```

Comment:
There is no previous OkEvent that is the cause of the problem. I replaced it with the InvokeEvent now, but not sure this is actually a nicer solution...